server/routes/appointments/appointmentsController.ts

```diff
@@ -746,14 +757,52 @@ export default class AppointmentsController {
     )
   }
 
-  async viewSubmittedPostSessionFeedback(req: Request, res: Response): Promise<void> {
-    const { user } = res.locals
-    const { accessToken } = user.token
+  async viewPostSessionFeedbackAsServiceProvider(req: Request, res: Response): Promise<void> {
+    const { accessToken } = res.locals.user.token
     const { actionPlanId, sessionNumber } = req.params
 
     const actionPlan = await this.interventionsService.getActionPlan(accessToken, actionPlanId)
     const referral = await this.interventionsService.getSentReferral(accessToken, actionPlan.referralId)
 
+    return this.viewSubmittedPostSessionFeedback(res, referral, actionPlan.id, sessionNumber, null, 'service-provider')
+  }
+
+  async viewPostSessionFeedbackAsProbationPractitioner(req: Request, res: Response): Promise<void> {
+    const { accessToken } = res.locals.user.token
+    const { referralId, sessionNumber } = req.params
+
+    const referral = await this.interventionsService.getSentReferral(accessToken, referralId)
+
+    if (referral.actionPlanId === null) {
```

Comment:
Seems that there's no harm having this check on the `viewPostSessionFeedbackAsServiceProvider` so can be moved into the private method `viewSubmittedPostSessionFeedback`?
If you agree that also perhaps have a single method like with `viewSupplierAssessmentFeedback`?

Reply:
I did initially try to re-use the function but I think my brain wasn't working too well yesterday. I've had another go and think it makes more sense now - I know you've already approved, but I don't suppose you'd mind taking another look as it'd be nice to get a second pair of eyes on the approach.